packages/core/src/web/app/views/beambox/ElementPanel/MainContent.tsx

```diff
@@ -0,0 +1,145 @@
+        shodowRef.current.style.display = 'none';
+      }
+
+      setTimeout(handleShadow, 500);
+    }
+  };
+
+  useEffect(() => {
+    setTimeout(() => {
+      // Display shadow after flaoting panel completly opened
+      waitFirstAnimation.current = false;
+    }, 500);
+  }, []);
+
+  useEffect(() => {
+    handleShadow();
+    // Check again to make sure all icons loaded
+    setTimeout(handleShadow, 300);
+    // eslint-disable-next-line hooks/exhaustive-deps
+  }, [contents]);
+
+  const isSearch = useMemo(() => contentType === ContentType.Search, [contentType]);
+  const isSearchEmpty = useMemo(() => {
+    if (!isSearch) return false;
+
+    if (!contents[0]) return true;
+
+    if (contents[0].loading || contents[0].fileNames?.length || contents[0].npIcons?.length) return false;
+
+    return true;
+  }, [isSearch, contents]);
+
+  return (
+    <div className={styles['shadow-container']}>
+      <div className={styles['scroll-content']} onScroll={handleShadow} ref={scrollDivRef}>
+        {!hasLogin && (
+          <ThemedButton
+            block
+            className={styles.login}
+            onClick={() => dialogCaller.showLoginDialog()}
+            theme="yellow"
+            type="primary"
+          >
+            {lang.sign_in}
+          </ThemedButton>
+        )}
+        {isSearch && (
+          <>
+            <ul className={styles.hints}>
+              <li>{lang.search_with_english}</li>
+              <li>{lang.search_examples} flower, school building.</li>
+            </ul>
+            {isSearchEmpty && (
+              <>
+                {contents[0]?.term && (
+                  <div className={styles.empty}>{sprintf(lang.search_no_results, contents[0].term)}</div>
+                )}
+                <div className={styles.bold}>{lang.browse_categories}</div>
+                <div className={styles.categories}>
+                  {types.map((key) => (
+                    <Button
+                      key={key}
+                      onClick={() => {
+                        setSearchKey(undefined);
+                        setActiveSubType(undefined);
+                        setActiveMainType(key);
+                      }}
+                      size="small"
+                      type="link"
+                    >
+                      {lang[key]}
+                    </Button>
+                  ))}
+                </div>
+              </>
+            )}
+          </>
+        )}
+        <div className={styles['icon-list']}>
+          {contentType === ContentType.MainType && historyIcons.length > 0 && (
+            <>
+              <div className={styles.subtitle}>{lang.recently_used}</div>
+              {historyIcons.map(({ npIcon, path }) =>
+                path ? (
+                  <BuiltinElement key={path.fileName} mainType={path.folder} path={path.fileName} />
+                ) : hasLogin ? (
+                  <NPElement icon={npIcon!} key={npIcon!.id} />
```

Comment:
Items may not be rendered if same item exist in `historyIcons`. Maybe due to key collision.